core/unit_test/TestGraph.hpp

```diff
@@ -792,32 +809,38 @@ struct GraphNodeTypes {
 
   // Type of an aggregate node.
   using aggregate_t = typename Kokkos::Impl::GraphImpl<Exec>::aggregate_impl_t;
+
+  // Type of a then node.
+  using then_t =
+      Kokkos::Impl::GraphNodeThenImpl<Exec, ThenFunctor<Kokkos::View<int>>>;
 };
 
 constexpr bool test_is_graph_kernel() {
   using types = GraphNodeTypes<TEST_EXECSPACE>;
   static_assert(Kokkos::Impl::is_graph_kernel_v<types::kernel_t>);
   static_assert(!Kokkos::Impl::is_graph_kernel_v<types::aggregate_t>);
+  static_assert(Kokkos::Impl::is_graph_kernel_v<types::then_t>,
+                "This should be verified until the 'then' has its own path to "
+                "the driver.");
   return true;
 }
 static_assert(test_is_graph_kernel());
 
 // This test checks the node types before/after a 'when_all'.
 TEST(TEST_CATEGORY, when_all_type) {
+  using types = GraphNodeTypes<TEST_EXECSPACE>;
```

Comment:
Still not convinced that changes to that unit tests improve anything,

Reply:
Let's keep them if I don't need to modify anything else, agreed ?

Reply:
Sure